spec/helpers/basespace_helper_spec.rb

```diff
@@ -0,0 +1,60 @@
+require "rails_helper"
+require "webmock/rspec"
+
+RSpec.describe BasespaceHelper, type: :helper do
+  describe "#files_for_basespace_dataset" do
+    context "#basespace API call returns successfully" do
+      before do
+        allow(HttpHelper).to receive(:get_json)
+          .and_return(
+            "Items" => [
+              {
+                "Name" => "Item 1",
+                "HrefContent" => "Download Path 1",
+                "Href" => "Source Path 1",
+                "Id" => "Id 1"
+              },
+              {
+                "Name" => "Item 2",
+                "HrefContent" => "Download Path 2",
+                "Href" => "Source Path 2",
+                "Id" => "Id 2"
+              }
+            ]
+          )
+      end
+
+      it "returns selected fields of files" do
+        files = helper.files_for_basespace_dataset("abc", "token")
+
+        expect(files).to eq(
+          [
+            {
+              name: "Item 1",
+              download_path: "Download Path 1",
+              source_path: "Source Path 1"
+            },
+            {
+              name: "Item 2",
+              download_path: "Download Path 2",
+              source_path: "Source Path 2"
+            }
+          ]
+        )
+      end
+    end
+
+    context "#basespace API call fails" do
```

Comment:
It looks like this test is more related to "basespace api returning zero results", correct?

Reply:
It's possible for the basespace API to fail as well. For example, with a 401 not authorized.

Returning zero results is a different error case that I didn't test. Just added tests for them.